client/asset/btc/btc.go

```diff
@@ -2311,7 +2317,10 @@ func (btc *baseWallet) fundInternal(keep uint64, minConfs uint32, lockUnspents b
 	}
 
 	if keep > 0 {
-		kept := leastOverFund(keep, utxos)
+		enough := func(_, sum uint64) (bool, uint64) {
+			return sum >= keep, 0
+		}
+		kept := leastOverFund(enough, utxos)
```

Comment:
Looks like you've shaddowed the `enough` provided as an input to `fundInternal`, so that `tryFund` a few lines down now gets this simplified enough func.

How about a constructor in coin_selection.go like:

```go
func reserveEnough(amt uint64) func(_, sum uint64) (bool, uint64) {
	return func(_, sum uint64) (bool, uint64) {
		return sum >= amt, 0
	}
}
```

so that it can be used inline here and in `estimateSwap` like

```go
kept := leastOverFund(reserveEnough(keep), utxos)`
```